### ISSUE TITLE
Poor instructions

### ISSUE THREAD

**Issue by MrAnonymouse** [MicrosoftDocs/OfficeDocs-o365seccomp#356]
Where do I get/what keys do I need/where exactly to put them in DNS settings?  These instructions are poorly written.  They are extremely vague and do not offer enough interlinking to Microsoft products to resolve the issue of "HOW" to setup DKIM from start to finish.  I would expect this type of vague information from Wikipedia not docs.microsoft.com

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 011db28a-84b4-a217-cda0-a04fe8178492
* Version Independent ID: 92f2667c-cf43-cbf4-22d1-7123520ef7a7
* Content: [Use DKIM to validate outbound email sent from your custom domain in Office 365](https://docs.microsoft.com/en-us/office365/securitycompliance/use-dkim-to-validate-outbound-email#EnableDKIMinO365)
* Content Source: [SecurityCompliance/use-dkim-to-validate-outbound-email.md](https://github.com/microsoftdocs/officedocs-o365seccomp/blob/public/SecurityCompliance/use-dkim-to-validate-outbound-email.md)
* Service: **o365-seccomp**
* GitHub Login: @KCCross
* Microsoft Alias: **krowley**

**Comment by denisebmsft** [comment]
@MrAnonymouse thank you so much for the feedback. We are sorry you had a difficult time with this, and we are looking into it. We'll post an update here soon.

**Comment by MSFTTracyP** [comment]
So the reason that the instructions are 'vague' is because there are _many_ domain registrars out there and since those companies are under their own ownership, Microsoft is not the best resource for guidance on how to navigate their systems. The company you use or used to register your DNS knows their system quite well, and you should contact them for directions while using their system or systems.

The initial domain used by Office365 (ex. <the name you chose>.onmicrosoft.com) has DKIM set up for it by default. Office 365 generates this automatically, so there is nothing more you need to do. If you're using this type of domain, your DKIM setup is already done. If you went to Office 365 and registered a custom domain bought from a Domain Registrar (a custom domain), you can also choose to do nothing about your DKIM since Office 365 creates a private and public key pair, enables DKIM signing, and then configures the Office 365 default policy for your custom domain. If you need to set up DMARC, or want more control of your DKIM domain, you will need to go to your Domain Registrar (from which you bought the domain name you use) and make **two CNAME entries per each custom domain that you've registered in Office 365** (so that Office 365 can round-robin between the two entries). 

You can find your domainGUID in your original custom domain registration. Find the MX record for your custom domain that ends in _.mail.protection.outlook.com_. The domainGUID is whatever comes before the '.mail' part of this MX record. Write the domainGUID down because you'll need it for the two CNAME entries you'll make. Also write down your initialDomain, which is the domain you created in Office365 at sign-up. For example litware.onmicrosoft.com (unless you're a GCC High customer - more on that in a moment).

Here's the format of the CNAME entries you will make using either selector1._domainkey or selector2._domainkey:

Host name:			selector1._domainkey
Points to address or value:	selector1-<**domainGUID**>._domainkey.<**initialDomain**> 
TTL:				3600

_**This format it different if you are a GCC High customer**_, so please see the note in the article [Use DKIM to validate outbound email sent from your custom domain in Office 365](https://docs.microsoft.com/en-us/office365/securitycompliance/use-dkim-to-validate-outbound-email#EnableDKIMinO365).

Here is an example of a domainGUID: **litware.com** vs. an initialDomain: **litware.onmicrosoft.com**

Follow the steps of your Registrar system to make the CNAME entries. Put the domainGUID in for <domainGUID> and <initialDomain> for initialDomain using the format sample above. Create two records, one for selector1 and one for selector2. Once this is complete, _Enable_ and _Test_ DKIM as per the steps recorded in this article: [Use DKIM to validate outbound email sent from your custom domain in Office 365](https://docs.microsoft.com/en-us/office365/securitycompliance/use-dkim-to-validate-outbound-email#EnableDKIMinO365).

I hope it clarifies how to setup DKIM from start to finish.
